src/main/java/org/alliancegenome/curation_api/dao/GafDAO.java

```diff
@@ -0,0 +1,67 @@
+package org.alliancegenome.curation_api.dao;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.persistence.Query;
+import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
+import org.alliancegenome.curation_api.model.entities.Gaf;
+import org.alliancegenome.curation_api.model.entities.Organization;
+import org.alliancegenome.curation_api.model.ingest.dto.GafDTO;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ApplicationScoped
+public class GafDAO extends BaseSQLDAO<Gaf> {
+
+	protected GafDAO() {
+		super(Gaf.class);
+	}
+
+	public Gaf persistGeneGoAssociation(Gaf gaf) {
+		String sql = """
+			insert into gene_go_annotation (id, gene_id,goterm_id)
+			VALUES (nextval('gene_go_annotation_seq'), :geneID, :goID)
+						""";
+		Query query = entityManager.createNativeQuery(sql);
+		query.setParameter("goID", gaf.getGoTerm().getId());
+		query.setParameter("geneID", gaf.getGene().getId());
+		query.executeUpdate();
+
+		sql = "select currval('gene_go_annotation_seq')";
+		Object object = entityManager.createNativeQuery(sql).getSingleResult();
+		gaf.setId((Long) object);
+		return gaf;
+	}
+
+	public Map<Long, GafDTO> getAllGafIdsPerProvider(Organization sourceOrganization) {
```

Comment:
Why not use the findByParams to do this?

Reply:
Because I want to keep the return very slim. If I got back the full object I would have to retrieve a full gene object and an Ontology object.

Reply:
There's a base `findIdsByParams` method to do that

Reply:
That method only retrieves the IDs. I need the IDs as well as the modentityIDs to cache that info .